packages/wordcount/src/index.js

```diff
@@ -111,18 +129,19 @@ function matchCharacters( text, regex, settings ) {
  */
 
 export function count( text, type, userSettings ) {
-	if ( '' === text ) {
-		return 0;
+	const settings = loadSettings( type, userSettings );
+	let matchRegExp;
+	if ( settings.type === 'words' ) {
+		matchRegExp = settings.wordsRegExp;
+		return countWords( text, matchRegExp, settings );
 	}
-
-	if ( text ) {
-		const settings = loadSettings( type, userSettings );
-		const matchRegExp = settings[ type + 'RegExp' ];
-		const results =
-			'words' === settings.type
-				? matchWords( text, matchRegExp, settings )
-				: matchCharacters( text, matchRegExp, settings );
-
-		return results ? results.length : 0;
+	if ( settings.type === 'characters_including_spaces' ) {
+		matchRegExp = settings.characters_including_spacesRegExp;
```

Comment:
Are this and `characters_excluding_spacesRegExp` below new settings introduced as of this pull request? I observe that they don't adhere to naming guidelines:

https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/#naming-conventions

I might expect something like:

```suggestion
		matchRegExp = settings.charactersIncludingSpacesRegExp;
```

Or perhaps more legible:

```suggestion
		matchRegExp = settings.charactersIncludingSpacesPattern;
```

**Edit:** Re-reading the changes, I think I understand this now as not being introduced here. I was trying to search for the literal phrase "characters_including_spacesRegExp", but I see it was previously concatenated above as `settings[ type + 'RegExp' ]`. The advice above still holds true, but given this isn't a change introduced here and for backwards-compatibility commitment, I don't think we want to do anything.

Reply:
Yeah… I almost refactored this portion, though `settings` de-facto being public interface stopped me. Maybe adding a comment makes sense, but I hope nobody will try to override this in the first place. Don't really want to make their lives easier and I would hope they re-consider :)